### ISSUE TITLE
pkg version: latest are now broken, appending <package>-latest to filename

### ISSUE THREAD

**Issue by jryberg** [saltstack/salt#42194]
### Description of Issue/Question
I'm trying to use a very simple state:
```
sl:
  pkg.installed:
    - version: latest
```
This does not work on Red Hat 6 but it works perfectly fine on Ubuntu 16.04

The minion tries to install package sl-latest instead of the latest version of sl

```
[user@hostname ~]$ salt-call -ldebug state.sls sl
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Connecting to master. Attempt 1 of 1
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506')
[DEBUG   ] Generated random reconnect delay between '1000ms' and '11000ms' (2725)
[DEBUG   ] Setting zmq_reconnect_ivl to '2725ms'
[DEBUG   ] Setting zmq_reconnect_ivl_max to '11000ms'
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506', 'clear')
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] SaltEvent PUB socket URI: /var/run/salt/minion/minion_event_b7a7406eda_pub.ipc
[DEBUG   ] SaltEvent PULL socket URI: /var/run/salt/minion/minion_event_b7a7406eda_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/minion/minion_event_b7a7406eda_pull.ipc
[DEBUG   ] Sending event: tag = salt/auth/creds; data = {'_stamp': '2017-07-07T08:00:33.968705', 'creds': {'publish_port': 4505, 'aes': '###REDACTED###', 'master_uri': 'tcp://salt:4506'}, 'key': ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506')}
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Determining pillar cache
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506')
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded state.sls
[DEBUG   ] LazyLoaded saltutil.is_running
[DEBUG   ] LazyLoaded grains.get
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506')
[DEBUG   ] Determining pillar cache
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506')
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[INFO    ] Loading fresh modules for state activity
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] In saltenv 'base', looking at rel_path 'sl.sls' to resolve 'salt://sl.sls'
[DEBUG   ] In saltenv 'base', ** considering ** path '/var/cache/salt/minion/files/base/sl.sls' to resolve 'salt://sl.sls'
[DEBUG   ] compile template: /var/cache/salt/minion/files/base/sl.sls
[DEBUG   ] Jinja search path: ['/var/cache/salt/minion/files/base']
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/sl.sls' using 'jinja' renderer: 0.00310182571411
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/sl.sls:
sl:
  pkg.installed:
    - version: latest

[DEBUG   ] LazyLoaded config.get
[DEBUG   ] Results of YAML rendering:
OrderedDict([('sl', OrderedDict([('pkg.installed', [OrderedDict([('version', 'latest')])])]))])
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/sl.sls' using 'yaml' renderer: 0.0039541721344
[DEBUG   ] LazyLoaded pkg.install
[DEBUG   ] LazyLoaded pkg.installed
[DEBUG   ] LazyLoaded boto.assign_funcs
[DEBUG   ] Module PSGet: Only available on Windows systems
[DEBUG   ] boto_kms requires boto 2.38.0.
[DEBUG   ] You should upgrade pyOpenSSL to at least 0.14.1 to enable the use of X509 extensions in the tls module
[DEBUG   ] Module DSC: Only available on Windows systems
[DEBUG   ] Could not LazyLoad pkg.ex_mod_init: 'pkg.ex_mod_init' is not available.
[INFO    ] Running state [sl] at time 04:00:35.631905
[INFO    ] Executing state pkg.installed for sl
[INFO    ] Executing command ['rpm', '-qa', '--queryformat', '%{NAME}_|-%{EPOCH}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-(none)\n'] in directory '/root'
[DEBUG   ] Could not LazyLoad pkg.check_db: 'pkg.check_db' is not available.
[INFO    ] Executing command ['yum', '--quiet', 'clean', 'expire-cache'] in directory '/root'
[DEBUG   ] output:
[INFO    ] Executing command ['yum', '--quiet', 'check-update'] in directory '/root'
[INFO    ] Executing command ['yum', '--quiet', 'list', 'available', 'sl'] in directory '/root'
[DEBUG   ] Failed to get holds, versionlock plugin is probably not installed
[INFO    ] Executing command ['yum', '-y', 'install', 'sl-latest'] in directory '/root'
[ERROR   ] Command '['yum', '-y', 'install', 'sl-latest']' failed with return code: 1
[ERROR   ] stdout: Loaded plugins: amazon-id, rhui-lb, security
Setting up Install Process
No package sl-latest available.
Error: Nothing to do
[ERROR   ] retcode: 1
[INFO    ] Executing command ['rpm', '-qa', '--queryformat', '%{NAME}_|-%{EPOCH}_|-%{VERSION}_|-%{RELEASE}_|-%{ARCH}_|-(none)\n'] in directory '/root'
[ERROR   ] Error occurred installing package(s). Additional info follows:

errors:
    - Loaded plugins: amazon-id, rhui-lb, security
      Setting up Install Process
      No package sl-latest available.
      Error: Nothing to do
[INFO    ] Completed state [sl] at time 04:00:44.238699 duration_in_ms=8606.794
[DEBUG   ] File /var/cache/salt/minion/accumulator/37080592 does not exist, no need to cleanup.
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'hostname', 'tcp://salt:4506')
[DEBUG   ] LazyLoaded highstate.output
```

The same state works perfectly fine on Ubuntu 16.04 where the latest sl package was installed. Using Ubuntu it did *not* add -latest to the filename. 

### Steps to Reproduce Issue
Install Red Hat 6
Run the following state:
```
sl:
  pkg.installed:
    - version: latest
```
#### Master
```
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: 2.2.1
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-1020-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```

#### Red Hat minion
```
Salt Version:
           Salt: 2016.11.6

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3c1
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, May 22 2015, 08:34:51)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.7 Santiago
        machine: x86_64
        release: 2.6.32-573.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.7 Santiago

```

**Comment by jryberg** [comment]
It does work with this configuration
```
sl:
  pkg.latest: []

```
But it's not the real point, the issue are affecting much bigger states that we are using with pillar data and a lot of dynamic states and the report is just the boil down of the issue.

**Comment by gtmanfred** [comment]
I am unable to get this to work on 16.04.

```
root@saltub:~# cat /srv/salt/test.sls
sl:
  pkg.installed:
    - version: latest
```
```
local:
----------
          ID: sl
    Function: pkg.installed
      Result: False
     Comment: Problem encountered installing package(s). Additional info follows:

              errors:
                  - Running scope as unit run-r7cad94be7cca4c309ff1c6c572ca01cb.scope.
                    E: Version 'latest' for 'sl' was not found
     Started: 16:04:50.800732
    Duration: 2182.085 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.182 s
```

**Comment by jryberg** [comment]
Hi,

I am able to install using latest from Ubuntu 16.04 hosts.

```
[root@hostname (minion) ~]# salt-call -ldebug state.sls sl
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Please install 'virt-what' to improve results of the 'virtual' grain.
[DEBUG   ] Connecting to master. Attempt 1 of 1
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506')
[DEBUG   ] Generated random reconnect delay between '1000ms' and '11000ms' (9347)
[DEBUG   ] Setting zmq_reconnect_ivl to '9347ms'
[DEBUG   ] Setting zmq_reconnect_ivl_max to '11000ms'
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506', 'clear')
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] SaltEvent PUB socket URI: /var/run/salt/minion/minion_event_88848c0314_pub.ipc
[DEBUG   ] SaltEvent PULL socket URI: /var/run/salt/minion/minion_event_88848c0314_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/minion/minion_event_88848c0314_pull.ipc
[DEBUG   ] Sending event: tag = salt/auth/creds; data = {'_stamp': '2017-07-10T08:25:48.357538', 'creds': {'publish_port': 4505, 'aes': '<REDACTED>', 'master_uri': 'tcp://salt:4506'}, 'key': ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506')}
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Determining pillar cache
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506')
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded state.sls
[DEBUG   ] LazyLoaded saltutil.is_running
[DEBUG   ] LazyLoaded grains.get
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506')
[DEBUG   ] Determining pillar cache
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506')
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[INFO    ] Loading fresh modules for state activity
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] In saltenv 'base', looking at rel_path 'sl.sls' to resolve 'salt://sl.sls'
[DEBUG   ] In saltenv 'base', ** considering ** path '/var/cache/salt/minion/files/base/sl.sls' to resolve 'salt://sl.sls'
[DEBUG   ] Fetching file from saltenv 'base', ** attempting ** 'salt://sl.sls'
[DEBUG   ] No dest file found
[INFO    ] Fetching file from saltenv 'base', ** done ** 'sl.sls'
[DEBUG   ] compile template: /var/cache/salt/minion/files/base/sl.sls
[DEBUG   ] Jinja search path: ['/var/cache/salt/minion/files/base']
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/sl.sls' using 'jinja' renderer: 0.00275301933289
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/sl.sls:
sl:
  pkg.installed:
    - version: latest

[DEBUG   ] LazyLoaded config.get
[DEBUG   ] Results of YAML rendering:
OrderedDict([('sl', OrderedDict([('pkg.installed', [OrderedDict([('version', 'latest')])])]))])
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/sl.sls' using 'yaml' renderer: 0.00320196151733
[DEBUG   ] LazyLoaded pkg.install
[DEBUG   ] LazyLoaded pkg.installed
[DEBUG   ] Trying paths: ['/root/.docker/config.json', '/root/.dockercfg']
[DEBUG   ] No config file found
[DEBUG   ] "GET /version HTTP/1.1" 200 223
[DEBUG   ] Attempting to run docker-py's "version" function with args=() and kwargs={}
[DEBUG   ] "GET /v1.27/version HTTP/1.1" 200 223
[DEBUG   ] Module DSC: Only available on Windows systems
[DEBUG   ] Module PSGet: Only available on Windows systems
[DEBUG   ] Could not LazyLoad pkg.ex_mod_init: 'pkg.ex_mod_init' is not available.
[INFO    ] Running state [sl] at time 08:25:49.074251
[INFO    ] Executing state pkg.installed for sl
[INFO    ] Executing command ['dpkg-query', '--showformat', '${Status} ${Package} ${Version} ${Architecture}\n', '-W'] in directory '/root'
[INFO    ] Executing command ['apt-get', '-q', 'update'] in directory '/root'
[INFO    ] Executing command ['apt-cache', '-q', 'policy', 'sl'] in directory '/root'
[DEBUG   ] Could not LazyLoad pkg.normalize_name: 'pkg.normalize_name' is not available.
[DEBUG   ] Could not LazyLoad pkg.check_db: 'pkg.check_db' is not available.
[DEBUG   ] Could not LazyLoad pkg.normalize_name: 'pkg.normalize_name' is not available.
[INFO    ] Executing command ['dpkg', '--get-selections', '*'] in directory '/root'
[INFO    ] Executing command ['systemd-run', '--scope', 'apt-get', '-q', '-y', '-o', 'DPkg::Options::=--force-confold', '-o', 'DPkg::Options::=--force-confdef', 'install', 'sl=3.03-17build1'] in directory '/root'
[INFO    ] Executing command ['dpkg-query', '--showformat', '${Status} ${Package} ${Version} ${Architecture}\n', '-W'] in directory '/root'
[DEBUG   ] Could not LazyLoad pkg.version_clean: 'pkg.version_clean' is not available.
[INFO    ] Made the following changes:
'sl' changed from 'absent' to '3.03-17build1'

[DEBUG   ] Refreshing modules...
[INFO    ] Loading fresh modules for state activity
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded saltutil.refresh_modules
[DEBUG   ] LazyLoaded event.fire
[DEBUG   ] SaltEvent PUB socket URI: /var/run/salt/minion/minion_event_88848c0314_pub.ipc
[DEBUG   ] SaltEvent PULL socket URI: /var/run/salt/minion/minion_event_88848c0314_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/minion/minion_event_88848c0314_pull.ipc
[DEBUG   ] Sending event: tag = module_refresh; data = {'_stamp': '2017-07-10T08:25:51.869383'}
[INFO    ] Completed state [sl] at time 08:25:51.869797 duration_in_ms=2795.546
[DEBUG   ] File /var/cache/salt/minion/accumulator/140254115725136 does not exist, no need to cleanup.
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'minion', 'tcp://salt:4506')
[DEBUG   ] LazyLoaded highstate.output
[DEBUG   ] LazyLoaded nested.output
local:
----------
          ID: sl
    Function: pkg.installed
      Result: True
     Comment: The following packages were installed/updated: sl=3.03-17build1
     Started: 08:25:49.074251
    Duration: 2795.546 ms
     Changes:
              ----------
              sl:
                  ----------
                  new:
                      3.03-17build1
                  old:

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   2.796 s
```

Ubuntu host versions-report
```
[root@hostname ~]# salt-minion --versions-report
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: 2.2.1
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-1020-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by jryberg** [comment]
According to the documentation it should work: https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkg.html

> If the version given is the string latest, the latest available package version will be installed à la pkg.latest.

**Comment by idokaplan** [comment]
Hi,

I'm having the same issue since upgrading to "2016.11.6".
Is there any update about this issue?

Thanks!
Ido

**Comment by terminalmage** [comment]
This isn't specific to RHEL6, it happens because of how this feature was implemented. It works when the `latest` keyword is used in the `pkgs` argument, but not when just a package name and version are passed.

I've opened https://github.com/saltstack/salt/pull/42275 to fix this, but in the meantime you can work around this by using the `pkgs` argument, like so:

```
sl:
  pkg.installed:
    - pkgs:
      - sl: latest
```

**Comment by idokaplan** [comment]
Hi,

Is this bug started since 2016.11.6 or previous versions?

Will it be fixed in the next release?

Ido

**Comment by terminalmage** [comment]
2016.11.6 is the first version with this bug. But, it's also the first version where the `latest` keyword is supported in the `pkg.installed` state.

I'm not sure yet what releases will contain a fix for this, since I just submitted the pull request that fixes it. But as I stated in my earlier comment, there is an easy workaround.

**Comment by idokaplan** [comment]
This is incorrect.
I'm using "latest" for a least 6 months and the workaround means that I need to update a lot of states.

Please advise.

Ido

**Comment by YuvalItzchakov** [comment]
I can concur what Ido is saying. We've been using "latest" for quite a while now, this is definitely not something new. From the documentation of 2016.3.2, pkg.installed:  **"If the version given is the string latest, the latest available package version will be installed à la pkg.latest."**

**Comment by jryberg** [comment]
We have been using "latest" for some time now, probably 6 month as I recall. 

We have our entire production environment depending on this since we are using pillar data to control versions but as default we are using "latest" if the pillar data is unset. 

@terminalmage, I agree the work around is "easy", but we have very many states depending on this logic so I'm afraid this work around are useless. 

Our salt environment are currently broken

**Comment by terminalmage** [comment]
Yeah, looking at the code, `latest` was there before, but changes were made for 2016.11.6 to make it work with state aggregation.

@jryberg "useless" is a strong word. You're welcome to roll back to an earlier release if it's too much work to implement the workaround (which you and I both agree is "easy"). I'm not sure what more you want, I submitted a pull request to fix it within an hour of learning of the problem.

**Comment by jryberg** [comment]
@terminalmage I'm sorry if I offended you but for us is the work around useless since we need to do changes in production code or revert back to an earlier version of the minion.  I'm sure it works perfectly fine and as you said, there is nothing else to do. Let's move forward instead =)

You did work fast, that is much appreciated! =) 

Maybe I see you at SaltConf17? 

Regards, Johan

**Comment by idokaplan** [comment]
Hi,

I have used to always install salt-minion latest version when provisioning a new server, and this bug enforce me to change it to install specific version.

Is there a repository that I can use that contain all salt-minion version?
Currently, I use this repo and only latest version package is available.

[saltstack-repo]
name=SaltStack repo for RHEL/CentOS 6
baseurl=https://repo.saltstack.com/yum/redhat/6/$basearch/latest
enabled=1
gpgcheck=1
gpgkey=https://repo.saltstack.com/yum/redhat/6/$basearch/latest/SALTSTACK-GPG-KEY.pub

Please advise.
Thanks,
Ido

**Comment by jryberg** [comment]
@idokaplan 
You can just change baseurl from latest to any released version.

If you browse to: https://repo.saltstack.com/yum/redhat/6/x86_64/ you will find all versions including the archive. 

For the later versions you will most likely be able to use the old gpgkey, but be aware you might point to another one as well. 

It's not one single repository that you asked for, but you should be able to write a state for your repo configurations and that will make this easy as well =)

Regards, Johan

**Comment by cachedout** [comment]
Closed via #42275
